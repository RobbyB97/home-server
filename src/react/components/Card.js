import React from 'react'
import {Link} from 'react-router-dom'

const Card = ({image, title, subtext, link}) => (
    <div>
        <div className="card__wrap">
            <a className="card__link" href={link}>
                <div className="card">
                    <img src={image}/>
                    <div className="card__text">
                        <h2>{title}</h2>
                        <p>{subtext}</p>
                    </div>
                </div>
            </a>
        </div>
    </div>
)

export default Card