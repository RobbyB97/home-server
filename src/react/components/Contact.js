import React from 'react';


export class Contact extends React.Component {
    render() {
        return (
            <section id="Contact" className="contact">
                <div className="contact__shape one"></div>
                <div className="contact__shape two"></div>
                Contact component
            </section>
        );
    }
}


export default Contact;