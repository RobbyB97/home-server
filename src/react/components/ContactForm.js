/* Libraries */
import React from 'react'


export class ContactForm extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            nameValue: ''
        };

        this.handleNameChange = this.handleNameChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleNameChange(event) {
        this.setState({
            ...this.state,
            nameValue: event.target.value
        });
    }

    handleSubmit(event) {
        console.log('hi')
    }

    render() {
        return (
            <div className="ContactForm">
                <br /><br /><br /><br /><br />
                <form id="contactForm">
                    <p>
                        <label>Name</label>
                        <input type="text" 
                            value={this.state.nameValue} 
                            onChange={this.handleNameChange} 
                            required 
                        />
                    </p>
                </form>
            </div>
        );
    }
}


export default ContactForm