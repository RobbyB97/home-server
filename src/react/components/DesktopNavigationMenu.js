import React from 'react'
import {NavLink} from 'react-router-dom'
import { connect } from 'react-redux'

import {setTitle} from '../actions/title'


const DesktopNavigationMenu = ({dispatch}) => (
    <div id="desktop-navigation-menu">
        <ul>
            <li>
                <NavLink to="/react/" onClick={() => {
                    dispatch(setTitle(''))
                }}>Home</NavLink>
            </li>
            <li>
                <NavLink to="/react/style_guide" onClick={() => {
                    dispatch(setTitle('Style Guide'))
                }}>Style Guide</NavLink>
            </li>
        </ul>
    </div>
)

export default connect()(DesktopNavigationMenu)