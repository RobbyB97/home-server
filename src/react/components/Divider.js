import React from 'react';


export class Divider extends React.Component {
    constructor(props) {
        super(props);
    }


    render() {
        return (
            <div className={`divider ${this.props.section}`}>
                <div className="divider__shape one"></div>
                <div className="divider__shape two"></div>
            </div>
        );
    }
}


export default Divider;