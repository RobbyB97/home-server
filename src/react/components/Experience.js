import React from 'react';


export class Experience extends React.Component {
    render() {
        return (
            <section id="Experience" className="experience">
                <div className="experience__shape one"></div>
                <div className="experience__shape two"></div>
                Experience component
            </section>
        );
    }
}


export default Experience;