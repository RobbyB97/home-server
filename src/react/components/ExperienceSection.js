import React from 'react';


export class ExperienceSection extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            active: false
        };
    }

    toggle = () => {
        this.setState({
            active: !this.state.active
        });
    };

    render() {
        return (
            <article className="experienceSection">
                <header className="experienceSection__header">
                    {this.props.logo &&
                        <img 
                            className="experienceSection__logo"
                            src={this.props.logo}
                        />
                    }

                    <div className="experienceSection__info">
                        <h2 className="experienceSection__company">
                            {this.props.company}
                        </h2>

                        {this.props.jobTitle &&
                            <h4 className="experienceSection__jobTitle">
                                {this.props.jobTitle}
                            </h4>
                        }

                        {this.props.dates &&
                            <h5 className="experienceSection__dates">
                                {this.props.dates[0]} {this.props.dates[1] && `- ${this.props.dates[1]}`}
                            </h5>
                        }
                    </div>
                </header>
                
                <section 
                    className="experienceSection__description"
                    data-active={this.state.active}
                >
                    <p>{this.props.description}</p>
                </section>

                <footer 
                    className="experienceSection__footer"
                    data-active={this.state.active}
                    onClick={this.toggle}
                >
                    <p>{this.state.active ? "⮝" : "⮟"}</p>
                </footer>
            </article>
        );
    }
}


export default ExperienceSection;