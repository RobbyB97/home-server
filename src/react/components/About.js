import React from 'react';

import Divider from './Divider'


export class About extends React.Component {
    render() {
        return (
            <section id="About" className="about">

                <div className="about__content one">
                    <div className="about__portrait">
                        <img src="/dist/images/aboutImage.png" />
                        <div className="about__text one">
                            <p>
                                <span className="one">Hi, my name's Robby. <span className="two">I'm a front-end developer and digital artist</span>.</span>
                            </p>
                        </div>
                    </div>
                </div>

                <Divider section="about two" />

                <div className="about__content two">
                    <div className="about__text two">
                        <p>
                            I want to help you succeed by building a beautiful, fast and functional website for your business.
                        </p>
                    </div>
                </div>
            </section>
        );
    }
}


export default About;