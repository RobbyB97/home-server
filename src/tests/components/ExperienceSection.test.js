import React from 'react';
import {shallow} from 'enzyme';

import {ExperienceSection} from '../../react/components/ExperienceSection';
import experienceSections from '../fixtures/experienceSections';


test('Render ExperienceSection without props', () => {
    const wrapper = shallow(<ExperienceSection />);
    expect(wrapper).toMatchSnapshot();
});

test('Render ExperienceSection with logo', () => {
    const wrapper = shallow(<ExperienceSection {...experienceSections[0]}/>);
    expect(wrapper).toMatchSnapshot();
});

test('Render ExperienceSection with company', () => {
    const wrapper = shallow(<ExperienceSection {...experienceSections[1]}/>);
    expect(wrapper).toMatchSnapshot();
});

test('Render ExperienceSection with job title', () => {
    const wrapper = shallow(<ExperienceSection {...experienceSections[2]}/>);
    expect(wrapper).toMatchSnapshot();
});

test('Render ExperienceSection with description', () => {
    const wrapper = shallow(<ExperienceSection {...experienceSections[3]}/>);
    expect(wrapper).toMatchSnapshot();
});

test('Render ExperienceSection with start date', () => {
    const wrapper = shallow(<ExperienceSection {...experienceSections[4]}/>);
    expect(wrapper).toMatchSnapshot();
});

test('Render ExperienceSection with start and end date', () => {
    const wrapper = shallow(<ExperienceSection {...experienceSections[5]}/>);
    expect(wrapper).toMatchSnapshot();
});

test('Render ExperienceSection with all props', () => {
    const wrapper = shallow(<ExperienceSection {...experienceSections[6]}/>);
    expect(wrapper).toMatchSnapshot();
});
