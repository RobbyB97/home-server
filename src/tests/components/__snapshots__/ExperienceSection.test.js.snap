// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render ExperienceSection with all props 1`] = `
<article
  className="experienceSection"
>
  <header
    className="experienceSection__header"
  >
    <img
      className="experienceSection__logo"
      src="Logo"
    />
    <div
      className="experienceSection__info"
    >
      <h2
        className="experienceSection__company"
      >
        Company
      </h2>
      <h4
        className="experienceSection__jobTitle"
      >
        jobTitle
      </h4>
    </div>
  </header>
  <section
    className="experienceSection__description"
    data-active={false}
  >
    <p>
      Description
    </p>
  </section>
  <footer
    className="experienceSection__footer"
    data-active={false}
    onClick={[Function]}
  >
    <p>
      ⮟
    </p>
  </footer>
</article>
`;

exports[`Render ExperienceSection with company 1`] = `
<article
  className="experienceSection"
>
  <header
    className="experienceSection__header"
  >
    <div
      className="experienceSection__info"
    >
      <h2
        className="experienceSection__company"
      >
        Company
      </h2>
    </div>
  </header>
  <section
    className="experienceSection__description"
    data-active={false}
  >
    <p />
  </section>
  <footer
    className="experienceSection__footer"
    data-active={false}
    onClick={[Function]}
  >
    <p>
      ⮟
    </p>
  </footer>
</article>
`;

exports[`Render ExperienceSection with description 1`] = `
<article
  className="experienceSection"
>
  <header
    className="experienceSection__header"
  >
    <div
      className="experienceSection__info"
    >
      <h2
        className="experienceSection__company"
      />
    </div>
  </header>
  <section
    className="experienceSection__description"
    data-active={false}
  >
    <p>
      Description
    </p>
  </section>
  <footer
    className="experienceSection__footer"
    data-active={false}
    onClick={[Function]}
  >
    <p>
      ⮟
    </p>
  </footer>
</article>
`;

exports[`Render ExperienceSection with job title 1`] = `
<article
  className="experienceSection"
>
  <header
    className="experienceSection__header"
  >
    <div
      className="experienceSection__info"
    >
      <h2
        className="experienceSection__company"
      />
      <h4
        className="experienceSection__jobTitle"
      >
        jobTitle
      </h4>
    </div>
  </header>
  <section
    className="experienceSection__description"
    data-active={false}
  >
    <p />
  </section>
  <footer
    className="experienceSection__footer"
    data-active={false}
    onClick={[Function]}
  >
    <p>
      ⮟
    </p>
  </footer>
</article>
`;

exports[`Render ExperienceSection with logo 1`] = `
<article
  className="experienceSection"
>
  <header
    className="experienceSection__header"
  >
    <img
      className="experienceSection__logo"
      src="Logo"
    />
    <div
      className="experienceSection__info"
    >
      <h2
        className="experienceSection__company"
      />
    </div>
  </header>
  <section
    className="experienceSection__description"
    data-active={false}
  >
    <p />
  </section>
  <footer
    className="experienceSection__footer"
    data-active={false}
    onClick={[Function]}
  >
    <p>
      ⮟
    </p>
  </footer>
</article>
`;

exports[`Render ExperienceSection with start and end date 1`] = `
<article
  className="experienceSection"
>
  <header
    className="experienceSection__header"
  >
    <div
      className="experienceSection__info"
    >
      <h2
        className="experienceSection__company"
      />
      <h5
        className="experienceSection__dates"
      >
        startDate
         
        - endDate
      </h5>
    </div>
  </header>
  <section
    className="experienceSection__description"
    data-active={false}
  >
    <p />
  </section>
  <footer
    className="experienceSection__footer"
    data-active={false}
    onClick={[Function]}
  >
    <p>
      ⮟
    </p>
  </footer>
</article>
`;

exports[`Render ExperienceSection with start date 1`] = `
<article
  className="experienceSection"
>
  <header
    className="experienceSection__header"
  >
    <div
      className="experienceSection__info"
    >
      <h2
        className="experienceSection__company"
      />
      <h5
        className="experienceSection__dates"
      >
        startDate
         
      </h5>
    </div>
  </header>
  <section
    className="experienceSection__description"
    data-active={false}
  >
    <p />
  </section>
  <footer
    className="experienceSection__footer"
    data-active={false}
    onClick={[Function]}
  >
    <p>
      ⮟
    </p>
  </footer>
</article>
`;

exports[`Render ExperienceSection without props 1`] = `
<article
  className="experienceSection"
>
  <header
    className="experienceSection__header"
  >
    <div
      className="experienceSection__info"
    >
      <h2
        className="experienceSection__company"
      />
    </div>
  </header>
  <section
    className="experienceSection__description"
    data-active={false}
  >
    <p />
  </section>
  <footer
    className="experienceSection__footer"
    data-active={false}
    onClick={[Function]}
  >
    <p>
      ⮟
    </p>
  </footer>
</article>
`;
